tilewe/tournament.py

```diff
@@ -90,67 +94,77 @@ def total_engines(self) -> int:
     @property
     def win_rates(self) -> list[float]: 
         return [self.win_counts[i] / max(1, self.game_counts[i]) for i in range(self.total_engines)]
+
+    @property
+    def draw_rates(self) -> list[float]: 
+        return [self.draw_counts[i] / max(1, self.game_counts[i]) for i in range(self.total_engines)]
+
+    @property
+    def win_draw_rates(self) -> list[float]: 
+        return [(self.win_counts[i] + self.draw_counts[i]) / max(1, self.game_counts[i]) for i in range(self.total_engines)]
+
+    @property
+    def lose_counts(self) -> int:
+        return [self.game_counts[i] - self.win_counts[i] - self.draw_counts[i] for i in range(self.total_engines)]
+
+    @property
+    def lose_rates(self) -> list[float]: 
+        return [self.lose_counts[i] / max(1, self.game_counts[i]) for i in range(self.total_engines)]
     
     @property
     def avg_scores(self) -> list[float]: 
         return [self.total_scores[i] / max(1, self.game_counts[i]) for i in range(self.total_engines)]
 
+    @property
+    def elo_delta(self) -> list[float]: 
+        return [self.elo_end[i] - self.elo_start[i] for i in range(self.total_engines)]
+
     @property
     def average_match_duration(self) -> float:
         return self.total_time / max(1, self.total_games)
     
     def get_matches_by_engine(self, engine: int) -> list[MatchData]:
-        filtered_matches = [x for x in self.match_data if engine in x.engines]
-        return filtered_matches
-    
-    def get_game_count_by_engine(self, engine: int) -> int:
-        return self.game_counts[engine]
-    
-    def get_wins_by_engine(self, engine: int) -> int:
-        return self.win_counts[engine]
-    
-    def get_win_rate_by_engine(self, engine: int) -> float:
-        return self.win_counts[engine] / max(1, self.game_counts[engine])
-    
-    def get_score_by_engine(self, engine: int) -> int:
-        return self.total_scores[engine]
-    
-    def get_avg_score_by_engine(self, engine: int) -> float:
-        return self.total_scores[engine] / max(1, self.game_counts[engine])
-    
-    def get_starting_elo_by_engine(self, engine: int) -> int:
-        return self.elo_start[engine]
-    
-    def get_ending_elo_by_engine(self, engine: int) -> int:
-        return self.elo_end[engine]
-    
-    def get_delta_elo_by_engine(self, engine: int) -> int:
-        return self.elo_end[engine] - self.elo_start[engine]
```

Comment:
These all have properties now, so just use `results.property[engine]` instead of results.function(engine)`